### PR TITLE
Separates the attachment field to a custom component

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
- Refactores also the 'search-value-selector.component'

### CODE REVIEW

File: client/src/app/shared/components/attachment-control/attachment-control.component.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    selector: 'os-attachment-control',
+    templateUrl: './attachment-control.component.html',
+    styleUrls: ['./attachment-control.component.scss']
+})
+export class AttachmentControlComponent implements OnInit, ControlValueAccessor {
+    /**
+     * Additional classes applied to the parent element
+     */
+    @Input()
+    public classes: string[];
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/attachment-control/attachment-control.component.ts]: Please remove, if not needed

File: client/src/app/site/agenda/components/topic-detail/topic-detail.component.html [hunk 2]
```diff
@@ -68,14 +68,7 @@ <h4 translate>Text</h4>
         </div>
 
         <!-- Attachments -->
-        <os-search-value-selector
-            ngDefaultControl
-            [form]="topicForm"
-            [formControl]="topicForm.get('attachments_id')"
-            [multiple]="true"
-            listname="{{ 'Attachments' | translate }}"
-            [InputListValues]="mediafilesObserver"
```
Comment (FinnStutzenstein) [inline client/src/app/site/agenda/components/topic-detail/topic-detail.component.html]: Please remove the mediafilesObserver from the component.

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html [hunk 3]
```diff
@@ -253,24 +251,14 @@ <h4>{{ 'Description' | translate }}:</h4>
 
             <!-- Attachments -->
             <div class="content-field" *ngIf="mediafilesAvailable">
```
Comment (FinnStutzenstein) [inline client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html]: remove the `*ngIf` and all mediafile related code (e.g. the property and the `mediafilesObserver` member) from the component.

File: client/src/app/shared/components/attachment-control/attachment-control.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { async, TestBed } from '@angular/core/testing';
+
+import { AttachmentControlComponent } from './attachment-control.component';
+
+fdescribe('AttachmentControlComponent', () => {
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/attachment-control/attachment-control.component.spec.ts]: please change this.

File: client/src/app/shared/components/attachment-control/attachment-control.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import { AttachmentControlComponent } from './attachment-control.component';
+
+describe('AttachmentControlComponent', () => {
+    // let component: AttachmentControlComponent;
+    // let fixture: ComponentFixture<AttachmentControlComponent>;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [AttachmentControlComponent]
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/attachment-control/attachment-control.component.spec.ts]: `imports: [E2EImportsModule]`

**Comment by tsiegleauq** [comment]
@GabrielInTheWorld the last merge brought up some conflicts

**Comment by tsiegleauq** [comment]
Rebased, solved conflicts, enabled the tests

**Pull request merged**